frame/collective/src/benchmarking.rs

```diff
@@ -34,6 +34,10 @@ fn assert_last_event<T: Config<I>, I: 'static>(generic_event: <T as Config<I>>::
 	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
 }
 
+fn id_to_remark_data<T: Config<I>, I: 'static>(id: u32, length: usize) -> SystemCall<T> {
+	SystemCall::<T>::remark { remark: vec![id.try_into().unwrap(); length] }
+}
```

Comment:
@ggwpez I don't think we can use the `remark` function for the proposal here. We have to create a valid `Call` here that is going to be the proposal in the benchmark, but we cannot use the `remark` function since it requires the argument to be a `Vec<u8>`.

Reply:
`id_to_remark_data(id: u32, len: usize) -> Vec<u8>` is what I meant.  
To test that it works you can increase the `MaxProposals` in the mock runtime to `512`.

Reply:
I don't think we can convert a `Vec<u8>` into a `Call`. The code expects a `Call`:
```rs
let proposal: T::Proposal = id_to_remark_data(i, length).into();
```
`T::Proposal` is actually `RuntimeCall`, so we cannot have `id_to_remark_data` return a `Vec<u8>`

Reply:
```rust
let proposal: T::Proposal = SystemCall::<T>::remark { remark: id_to_remark_data(i, length) }.into();
```
You can obviously think of another way as well.

Reply:
The problem here is that we cannot have two same proposals. The remark call accepts a `Vec<u8>`. In the benchmark code, we make proposals with the same call(`remark`) but use different arguments. We submit a vector with repeated values for a specified length. These values range from `0` to `T::MaxProposals::get()`. If `T::MaxProposals::get()` is greater than `255` we run into a problem, because the `remark` function accepts only `u8`s.
I currently have two ideas on how to solve this.

1. We can change the length of the vector if the `T::MaxProposals::get()` is greater than `255`. What I mean by this is the following: We generate `remark` function calls for `0..255` with the specified length, and after we reach `255` we can increase the length by one and repeat the process. This way the calls aren't going to be the same.

2. Currently we submit repeated values for a given length to the `remark` function. What we could do instead is not repeat the same value but have the call data generated like this: from `0` to `255` we will have the following vector: `[i, repeat 0s for the given length]`, and after we reach `255` we do the same thing but we have `i` at the next index. `[0, i, repeat 0s for the given length]`...

Or maybe I am missing something and overthinking this :P